### PR TITLE
feat(api-reference): SSR

### PR DESCRIPTION
**Pull request by hanspagel**
## Problem

you can use the Vue components to render something on the server-side
but it's not convenient, and not really framework-agnostic

## Test

```bash
pnpm turbo --filter @scalar/server-side-rendering dev
```

## Solution

this PR introduces a great SSR helper for a great DX, not tied to any framework :)

* new package `@scalar/server-side-rendering`
  * generates the HTML
  * inline CSS
  * adds a bit of JS to determine light/dark-mode client-side before rendering anything
  * does the client-hydration with bundled JS (to avoid a hydration mismatch)
  * can be used in any JS/TS-framework (hono, react… whatever)
* new package `@scalar/client-side-rendering` (used to be `@scalar/core`, streamlined the APIs for server- and client-side rendering)
* use `@scalar/client-side-rendeirng` in `@scalar/core` (same public API, but deprecated functions) 
* made `examples/vue-ssr` a playground for `@scalar/server-side-rendering`

**not part of this PR**

* no docs yet
* our integrations still use `@scalar/core`

```ts
import { renderApiReference, getJsAsset } from '@scalar/server-side-rendering'

// Render once at startup
const html = await renderApiReference({
  config: {
    url: 'https://registry.scalar.com/@scalar/apis/galaxy?format=json',
  },
  pageTitle: 'My API Reference',
})

const js = getJsAsset()

// Serve the JS bundle for client-side hydration
app.get('/scalar/scalar.js', (c) => {
  return c.body(js, {
    headers: { 'content-type': 'application/javascript' },
  })
})

// Serve the pre-rendered HTML
app.get('/scalar', (c) => c.html(html))
```

## Checklist

- [x] I explained why the change is needed.
- [x] I added a changeset. 
- [x] I added tests.
- [ ] I updated the documentation.

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it introduces new rendering packages and changes `@scalar/api-reference` hydration/app creation behavior, which can affect SSR output and client hydration correctness.
> 
> **Overview**
> **Introduces two new packages for framework-agnostic rendering.** `@scalar/client-side-rendering` generates a static HTML document that loads the API Reference from the CDN, while `@scalar/server-side-rendering` server-renders the Vue output, inlines CSS, emits a pre-paint color-mode script, and provides `getJsAsset()` for hydration via the bundled standalone JS.
> 
> **Refactors existing HTML rendering APIs.** `@scalar/core`’s HTML rendering helpers are now thin, deprecated wrappers around `@scalar/client-side-rendering`, and `@scalar/api-reference` drops the `./ssr` export while updating the standalone HTML API to hydrate server-rendered markup (using `createSSRApp`) to avoid duplicate app creation; tests were added to assert no hydration warnings.
> 
> **Repo plumbing updates.** Removes the `examples/vue-ssr` example and repurposes it as the `@scalar/server-side-rendering` playground, updates CI to run the new SSR playground job, and adjusts knip/lockfile/workspace metadata accordingly.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1c0a52f17da99f9a1586f5b46c54bc9086e00112. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/server-side-rendering/src/ssr.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  }
+
+  const jsonString = JSON.stringify(jsonProps)
+
+  if (functionProps.length === 0) {
+    return jsonString
+  }
+
+  // Merge: remove trailing } from JSON, append function props
+  return `${jsonString.slice(0, -1)}, ${functionProps.join(', ')}}`
```
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/server-side-rendering/src/ssr.ts]: kind of an edge case, a source/url/content attribute is required, but let's fix it though

**Comment by amritk** [comment]
@cursoragent can you add a test that checks the console any warnings/errors that contain the word hydration

**Comment by hanspagel** [comment]
@cursor push 05f55ac7b2

File: packages/server-side-rendering/src/ssr.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>${title}</title>
+    <style>${css}</style>
+  </head>
+  <body>
+    ${bodyScript}
+    <div id="app">${html}</div>
+    <script src="${cdn}"></script>
+    <script>Scalar.createApiReference('#app', ${configJs})</script>
```
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/server-side-rendering/src/ssr.ts:184]: yeah, I mean if the developer puts in a script, it's gonna render a script… it's fine I guess

**Comment by hanspagel** [comment]
we gonna merge and I ff with @amritk’s suggested changes next week
